src/ContactPeople/ContactPeopleView.js

```diff
@@ -1,86 +1,66 @@
-  }
-
-  componentDidUpdate(prevProps) {
-    const { initialValues } = this.props;
-
-    if (!isEqual(initialValues.contacts, prevProps.initialValues.contacts)) {
-      this.updateQueryContacts(initialValues.contacts);
-    }
-  }
-
-  updateQueryContacts = contacts => {
-    const { parentMutator } = this.props;
-
-    let newQuery = 'query=(id=null)';
-
-    if (contacts.length >= 1) {
-      const items = contacts.map(item => {
-        return `id="${item}"`;
-      });
-      const biuldQuery = items.join(' or ');
-
-      newQuery = `query=(${biuldQuery})`;
-    }
-
-    parentMutator.queryCustom.update({ contactIDs: newQuery });
-  }
-
-  render() {
-    const { parentResources } = this.props;
-
-    if (get(parentResources, 'contacts.isPending', true)) {
-      return <Icon icon="spinner-ellipsis" />;
-    }
-
-    const contacts = get(parentResources, 'contacts.records', []);
-    const categories = get(parentResources, 'vendorCategory.records', []);
-
-    if (!isEmpty(contacts)) {
-      return (
-        <div style={{ width: '100%' }}>
-          {
-            contacts.map(contact => (
-              <ContactPerson
-                contact={contact}
-                categories={categories}
-                key={contact.id}
-              />
-            ))
-          }
-        </div>
-      );
-    } else {
-      return (
-        <div>
-          <p>{<FormattedMessage id="ui-organizations.contactPeople.noContactAvailable" />}</p>
-        </div>
-      );
-    }
-  }
-}
+import { FormattedMessage } from 'react-intl';
+import {
+  find,
+  get,
+} from 'lodash';
+
+import {
+  Icon,
+  Layout,
+  MultiColumnList,
+} from '@folio/stripes/components';
+
+import { transformCategoryIdsToLables } from '../common/utils';
+
+const visibleColumns = ['name', 'categories', 'email', 'phone', 'icon'];
+const columnMapping = {
+  name: <FormattedMessage id="ui-organizations.contactPeople.name" />,
+  categories: <FormattedMessage id="ui-organizations.contactPeople.categories" />,
+  email: <FormattedMessage id="ui-organizations.contactPeople.email" />,
+  phone: <FormattedMessage id="ui-organizations.contactPeople.phone" />,
+  icon: null,
+};
+
+const ContactPeopleView = ({ categoriesDict, contacts, openContact }) => {
+  const resultsFormatter = {
+    name: ({ firstName, lastName }) => `${firstName} ${lastName}`,
+    categories: ({ categories = [] }) => transformCategoryIdsToLables(categoriesDict, categories),
+    email: ({ emails }) => get(find(emails, 'isPrimary'), 'value', ''),
+    phone: ({ phoneNumbers }) => get(find(phoneNumbers, 'isPrimary'), 'phoneNumber', ''),
+    icon: () => <Icon icon="caret-right" />,
+  };
+
+  return (
+    <Fragment>
+      <MultiColumnList
+        columnMapping={columnMapping}
+        contentData={contacts}
+        formatter={resultsFormatter}
+        onRowClick={openContact}
+        visibleColumns={visibleColumns}
+      />
+      {Boolean(contacts.length) && (
```

Comment:
there is no such requirement in the ticket UIORGS-122

Reply:
removed